Mappers/HttpMapper.php

```diff
@@ -1,165 +1,202 @@
+    public function update(array $objectData)
     {
-        $content = $this->getHttpClient()->sendHttpRequest($action, $url, $options)->getBody();
-        
-        if($content === null){
-            return true;
-        }
-
-        if(array_key_exists('items', $content)){
-            return $this->buildPaginationCollection($content);
-        }
-
-        return $this->buildObject($content);
+        return $this->sendHttpRequest(self::PUT, $this->urlPlaceholderRender->updateUrl($objectData), [], $objectData);
     }
 
     /**
-     * @deprecated
-     * @param $action
-     * @param $url
-     * @param $options
-     * @return ArraySerializable|\Modules\LibraryModule\Models\Collection|void
-     * @throws EnergonException
+     * @param string $byPropertyName
+     * @param $propertyValue
+     * @return bool
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    protected function sendHttpRequestForGetStatus($action, $url, $options)
+    public function delete(string $byPropertyName, $propertyValue): bool
     {
-        return $this->getHttpClient()->sendHttpRequestForGetStatus($action, $url, $options);
+        $this->sendHttpRequest(self::DELETE, $this->urlPlaceholderRender->deleteUrl([$byPropertyName => $propertyValue]));
+
+        return true;
     }
 
+
     /**
-     * @param ArraySerializable $object
-     * @param $identifierName
-     * @param $urlName
-     * @return string
+     * @param array $objectData
+     * @return Collection|mixed
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    protected function getUrlWithObjectAndIdentifier(ArraySerializable $object, $identifierName, $urlName)
+    public function updatePatch(array $objectData)
     {
-        $identifier = $object->toArray()[$identifierName];
-        return vsprintf($this->getHttpMapperConfig($urlName), [$identifier]);
+        return $this->sendHttpRequest(self::PATH, $this->urlPlaceholderRender->updateUrl($objectData), [], $objectData);
     }
 
     /**
-     * @param $identifierValue
+     * @param $method
      * @param $url
-     * @return string
+     * @param array $params
+     * @param $body
+     * @return Collection|mixed
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\IllegalHeaderValueException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    private function getUrlWithIdentifier($identifierValue, $url)
+    protected function sendHttpRequest($method, $url, $params = [], $body = null)
     {
-        $urlTemplate = $this->getHttpMapperConfig($url);
-        return vsprintf($urlTemplate, [$identifierValue]);
+        if (!$this->headersBuilder->hasHeader('Content-Type')) {
```

Comment:
In such implementation when builder is injected you can get temporary coupling between requests.

Reply:
fixed